core/cloudflow-maven-plugin/src/main/scala/com/lightbend/cloudflow/maven/CloudflowAggregator.scala

```diff
@@ -0,0 +1,171 @@
+        findBlueprint(project).orElse(acc)
+    }
+  }
+
+  def getStreamlets(allProjects: Seq[MavenProject], log: Log) = {
+    allProjects.foldLeft(Map.empty[String, Map[String, Config]]) {
+      case (acc, project) =>
+        try {
+
+          log.info(s"extracting streamlets $project")
+
+          val rawStreamlets =
+            FileUtil.readLines(new File(project.getBuild.getDirectory, Constants.STREAMLETS_FILE))
+
+          val streamlets: Map[String, Config] = rawStreamlets.map { k =>
+            val file = new File(project.getBuild.getDirectory, URLEncoder.encode(k, "UTF-8"))
+            val config = ConfigFactory
+              .parseFile(file)
+              .resolve()
+            k -> config
+          }.toMap
+
+          (acc + (project.getName -> streamlets))
+        } catch {
+          case ex: Throwable =>
+            log.warn(s"No streamlets found in project $project")
+            (acc)
+        }
+    }
+  }
+
+  def getImages(allProjects: Seq[MavenProject], log: Log) = {
+    allProjects.foldLeft(Map.empty[String, String]) {
+      case (acc, project) =>
+        try {
+
+          log.info(s"extracting image names $project")
+
+          val image =
+            FileUtil.readLines(new File(project.getBuild.getDirectory, Constants.DOCKER_IMAGE_FILE)).mkString.trim
+
+          (acc + (project.getName -> image))
+        } catch {
+          case ex: Throwable =>
+            log.warn(s"No images found in project $project")
+            (acc)
+        }
+    }
+  }
+
+  def getCR(topLevel: MavenProject): Option[CloudflowCR] = {
+    val crFile = new File(topLevel.getBuild.getDirectory, topLevel.getName + ".json")
+
+    if (crFile.exists()) {
+
+      val cr = getCR(
+        FileUtil
+          .readLines(crFile)
+          .mkString("\n"))
+
+      Some(cr)
+    } else {
+      None
+    }
+  }
+
+  def getCR(str: String): CloudflowCR = {
+    str.parseJson.convertTo[CloudflowCR]
+  }
+
+  private def readBlueprint(blueprintFile: Option[File]) = {
+    val blueprintStr = FileUtil.readLines(blueprintFile.get).mkString("")
+
+    (blueprintStr, ConfigFactory.parseString(blueprintStr).getObject("blueprint.streamlets").asScala.toMap)
+  }
+
+  def generateLocalCR(projectId: String, version: String, allProjects: Seq[MavenProject], log: Log) = {
+    val streamletsPerProject = CloudflowAggregator.getStreamlets(allProjects, log)
+
+    val streamlets = streamletsPerProject.foldLeft(Map.empty[String, Config]) { case (acc, (_, v)) => acc ++ v }
+
+    val (blueprintStr, blueprint) = readBlueprint(CloudflowAggregator.getBlueprint(allProjects, log))
+
+    val placeholderImages = blueprint.map { case (k, _) => k -> "placeholder" }
+
+    Generator.generate(
+      projectId = projectId,
+      version = version,
+      blueprintStr = blueprintStr,
+      streamlets = streamlets,
+      dockerImages = placeholderImages)
+  }
+
+  def generateCR(projectId: String, version: String, allProjects: Seq[MavenProject], log: Log) = {
```

Comment:
nitpick: function return type to be specified

Reply:
ok